src/AtomBeautify.ts

```diff
@@ -1,203 +1,334 @@
-        this.subscriptions.add(atom.commands.add("atom-workspace", "atom-beautify:help-debug-editor", this.debug.bind(this)));
-        this.subscriptions.add(atom.commands.add("atom-workspace", "atom-beautify:open-settings", this.openSettings.bind(this)));
-        this.subscriptions.add(atom.commands.add(".tree-view .file .name", "atom-beautify:beautify-file", this.beautifyFile.bind(this)));
-        this.subscriptions.add(atom.commands.add(".tree-view .directory .name", "atom-beautify:beautify-directory", this.beautifyDirectory.bind(this)));
-    }
+  private unibeautify: Unibeautify;
+  private subscriptions: CompositeDisposable;
 
-    public deactivate(): void {
-        return this.subscriptions.dispose();
-    }
+  // Activates the Atom Beautify package
+  public activate(): void {
+    logger.info("Atom Beautify activated with beautifiers", { beautifiers });
+    this.unibeautify = newUnibeautify();
+    this.unibeautify.loadBeautifiers(beautifiers);
+    this.subscriptions = new CompositeDisposable();
+    this.subscriptions.add(
+      this.handleSaveEvent(),
+      atom.commands.add(
+        "atom-workspace",
+        "atom-beautify:beautify-editor",
+        this.beautifyEditor.bind(this)
+      ),
+      atom.commands.add(
+        "atom-workspace",
+        "atom-beautify:help-debug-editor",
+        this.debug.bind(this)
+      ),
+      atom.commands.add(
+        "atom-workspace",
+        "atom-beautify:open-settings",
+        this.openSettings.bind(this)
+      ),
+      atom.commands.add(
+        ".tree-view .file .name",
+        "atom-beautify:beautify-file",
+        this.beautifyFile.bind(this)
+      ),
+      atom.commands.add(
+        ".tree-view .directory .name",
+        "atom-beautify:beautify-directory",
+        this.beautifyDirectory.bind(this)
+      ),
+      atom.config.onDidChange(
+        "atom-beautify.general.loggerLevel",
+        (values: {oldValue: string; newValue: string}) => {
+          logger.level = values.newValue;
+          logger.info(`Logger level changed from ${values.oldValue} to ${values.newValue}`);
+        },
+      ),
+    );
+  }
 
-    public get config() {
-        return _.merge(Config, require("./options.json"));
-    }
+  // Deactivates the Atom Beautify package
+  public deactivate(): void {
+    return this.subscriptions.dispose();
+  }
 
-    private handleSaveEvent() {
-      return atom.workspace.observeTextEditors((editor: Atom.TextEditor) => {
-        const onWillSaveHandler = ({path: filePath}: {
-          path: string
-        }) => this.beautifyOnSaveHandler({
-          filePath: filePath
-        }, editor);
-        return this.subscriptions.add(editor.getBuffer().onWillSave(onWillSaveHandler as any));
-      });
+  // Retrieve the configuration options for Atom Beautify to display in Atom Settings
+  public get config() {
+    return Object.assign(Config, require("./options.json"));
```

Comment:
`_.merge` is a recursive / deep merge of properties: https://lodash.com/docs/#merge

Using `Object.assign` is OK so long as `./options.json` does not contain a `general` property like `Config` does: https://github.com/Glavin001/atom-beautify/blob/3b4508e6fe1842e8058684a8edbd42f09bf9b8ba/src/config.ts#L2
I think this is true.

Reply:
Should `./options.json` be committed to Git?
It is for current Atom-Beautify: https://github.com/Glavin001/atom-beautify/blob/master/src/options.json

Reply:
It can be, but I don't see any reason why it should.  The file is autogenerated.

Reply:
It doesn't need to be committed but it does need to be bundled into the package when pushed to Atom package registry. It should not be generated on the user's machine during installation or otherwise.